File: SPCode/CS/Data/Support/SPMaster.cs
using System.Text.RegularExpressions;
using System.Diagnostics;
using System;
using System.Drawing.Drawing2D;
using System.Windows.Forms;
using System.Collections;
using System.Drawing;
using Microsoft.VisualBasic;
using System.Data.SqlClient;
using System.Data;
using System.Collections.Generic;
using WeifenLuo.WinFormsUI;
using Microsoft.Win32;
using WeifenLuo;

namespace SoftLogik.Win
{
	public class SPMaster
	{
		
		
		
		public static SPDataProxy.SPMasterDataTable GetMasterDetail(long MasterID)
		{
			SPDataProxy.SPMasterDataTable data;
			using (SPDataProxyTableAdapters.taSPMaster dataCmd = new SPDataProxyTableAdapters.taSPMaster())
			{
				data = dataCmd.GetMaster(MasterID);
			}
			
			
			return data;
		}
		
		public static SPDataProxy.SPMasterDataTable GetMaster(string TypeID)
		{
			SPDataProxy.SPMasterDataTable data;
			using (SPDataProxyTableAdapters.taSPMaster dataCmd = new SPDataProxyTableAdapters.taSPMaster())
			{
				data = dataCmd.GetMasterLookup(TypeID);
			}
			
			
			return data;
		}
		public static SPDataProxy.SPMasterDataTable GetEmptyMaster(string TypeID)
		{
			SPDataProxy.SPMasterDataTable data;
			using (SPDataProxyTableAdapters.taSPMaster dataCmd = new SPDataProxyTableAdapters.taSPMaster())
			{
				data = dataCmd.GetEmptyMaster(TypeID);
			}
			
			
			return data;
		}
		public static int InsertMaster(string Name, string Note, string TypeID)
		{
			int retval;
			using (SPDataProxyTableAdapters.taSPMaster dataCmd = new SPDataProxyTableAdapters.taSPMaster())
			{
				retval = dataCmd.Insert(Name, Note, TypeID);
			}
			
			
			return retval;
		}
		public static int UpdateMaster(long MasterID, string Name, string Note)
		{
			int retval;
			using (SPDataProxyTableAdapters.taSPMaster dataCmd = new SPDataProxyTableAdapters.taSPMaster())
			{
				retval = dataCmd.Update(MasterID, Name, Note);
			}
			
			
			return retval;
		}
		public static int DeleteMaster(long MasterID)
		{
			int retval;
			using (SPDataProxyTableAdapters.taSPMaster dataCmd = new SPDataProxyTableAdapters.taSPMaster())
			{
				retval = dataCmd.Delete(MasterID);
			}
			
			
			return retval;
		}
		
	}
	
}

File: SPCode/Data/Support/SPMaster.vb
Public Class SPMaster


    Public Shared Function GetMasterDetail(ByVal MasterID As Long) As SPDataProxy.SPMasterDataTable
        Dim data As SPDataProxy.SPMasterDataTable
        Using dataCmd As New SPDataProxyTableAdapters.taSPMaster
            data = dataCmd.GetMaster(MasterID)
        End Using

        Return data
    End Function

    Public Shared Function GetMaster(ByVal TypeID As String) As SPDataProxy.SPMasterDataTable
        Dim data As SPDataProxy.SPMasterDataTable
        Using dataCmd As New SPDataProxyTableAdapters.taSPMaster
            data = dataCmd.GetMasterLookup(TypeID)
        End Using

        Return data
    End Function
    Public Shared Function GetEmptyMaster(ByVal TypeID As String) As SPDataProxy.SPMasterDataTable
        Dim data As SPDataProxy.SPMasterDataTable
        Using dataCmd As New SPDataProxyTableAdapters.taSPMaster
            data = dataCmd.GetEmptyMaster(TypeID)
        End Using

        Return data
    End Function
    Public Shared Function InsertMaster(ByVal Name As String, ByVal Note As String, ByVal TypeID As String) As Integer
        Dim retval As Integer
        Using dataCmd As New SPDataProxyTableAdapters.taSPMaster
            retval = dataCmd.Insert(Name, Note, TypeID)
        End Using

        Return retval
    End Function
    Public Shared Function UpdateMaster(ByVal MasterID As Long, ByVal Name As String, ByVal Note As String) As Integer
        Dim retval As Integer
        Using dataCmd As New SPDataProxyTableAdapters.taSPMaster
            retval = dataCmd.Update(MasterID, Name, Note)
        End Using

        Return retval
    End Function
    Public Shared Function DeleteMaster(ByVal MasterID As Long) As Integer
        Dim retval As Integer
        Using dataCmd As New SPDataProxyTableAdapters.taSPMaster
            retval = dataCmd.Delete(MasterID)
        End Using

        Return retval
    End Function

End Class
